### PR TITLE
Finished core build

### PR DESCRIPTION
**Pull request by hoangphanusc**
- Finished building package interface
- Added error handling for all repositories
- Added package infrastructure and main.go

TODO: Set up postgresql server and run tests

### CODE REVIEW

File: interfaces/repositories.go [hunk 1]
```diff
@@ -35,7 +36,7 @@ func NewDbUserRepo(dbHandlers map[string]DbHandler) *DbUserRepo {
 }
 
 func (repo *DbUserRepo) Store(user usecases.User) error {
-	err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO users (id, user_name, player_id, personal_info)
+	_, err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO users (id, user_name, player_id, personal_info)
```
Comment (trietphm) [inline interfaces/repositories.go]: Insert `id`? Where do you get `id` from? Is it unique?

File: interfaces/repositories.go [hunk 2]
```diff
@@ -81,7 +120,7 @@ func NewDbPlayerRepo(dbHandlers map[string]DbHandler) *DbPlayerRepo {
 }
 
 func (repo *DbPlayerRepo) Store(player domain.Player) error {
-	err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO players (id, name)
+	_, err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO players (id, name)
```
Comment (trietphm) [inline interfaces/repositories.go]: Same https://github.com/frostablevenus/game-tracker/pull/2/files#r70233053

File: interfaces/repositories.go [hunk 3]
```diff
@@ -112,13 +151,13 @@ func NewDbLibraryRepo(dbHandlers map[string]DbHandler) *DbLibraryRepo {
 }
 
 func (repo *DbLibraryRepo) Store(library usecases.Library) error {
-	err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO libraries (id, player_id) VALUES ('%d', '%v')`,
+	_, err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO libraries (id, player_id) VALUES ('%d', '%v')`,
```
Comment (trietphm) [inline interfaces/repositories.go]: https://github.com/frostablevenus/game-tracker/pull/2/files#r70233053

File: interfaces/repositories.go [hunk 4]
```diff
@@ -175,7 +214,7 @@ func NewDbGameRepo(dbHandlers map[string]DbHandler) *DbGameRepo {
 }
 
 func (repo *DbGameRepo) Store(game domain.Game) error {
-	err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO games (id, name, producer value)
+	_, err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO games (id, name, producer value)
```
Comment (trietphm) [inline interfaces/repositories.go]: https://github.com/frostablevenus/game-tracker/pull/2/files#r70233053

File: main.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	"fmt"
+	"net/http"
+
+	"game-tracker/infrastructure"
+	"game-tracker/interfaces"
+	"game-tracker/usecases"
+)
+
+func main() {
+	dbAddress := "postgres://postgres:postgres5263@127.0.0.1:5432/gameTrackerDb?sslmode=disable"
```
Comment (trietphm) [inline main.go]: It's better to have a config struct

File: interfaces/repositories.go (hunk 1 above)
Comment (hoangphanusc) [inline interfaces/repositories.go]: Do I need to name the field user_id instead? (id is primary key of table "users")

File: main.go (hunk 5 above)
Comment (hoangphanusc) [inline main.go]: I'm planning to add config in next pr

File: interfaces/repositories.go [hunk 6]
```diff
@@ -73,6 +79,39 @@ func (repo *DbUserRepo) FindById(id int) (usecases.User, error) {
...
+	count := 0
+	for row.Next() {
+		count++
+	}
+	return count
+}
+
+func (repo *DbUserRepo) NameExisted(userName string) bool {
+	row, _ := repo.dbHandler.Query(fmt.Sprintf(`SELECT user_name FROM users
+		WHERE user_name='%s' LIMIT 1`, userName))
```
Comment (trietphm) [inline interfaces/repositories.go]: Be careful with SQL Injection! Try with this username=`' OR '1'='1`
Take a look:
- https://github.com/jmoiron/sqlx
- https://github.com/jinzhu/gorm

File: interfaces/repositories.go (hunk 1 above)
Comment (trietphm) [inline interfaces/repositories.go]: `AUTO INCREMENT`
http://www.tutorialspoint.com/postgresql/postgresql_using_autoincrement.htm

**Pull request closed without merging**